### PR TITLE
Remove now obsolete CellDataTransferBuffer

### PR DESCRIPTION
**Pull request by kronbichler**
Follow-up to #13802: Now we never use `CellDataTransferBuffer` any more, so we can remove it. I did not include it in the other PR because we had this in the public namespace, so in principle we would need to deprecate it before removing it. On the other hand, I cannot imagine a use case of this class outside of `GridTools`.

### CODE REVIEW

File: include/deal.II/grid/grid_tools.h [hunk 1]
```diff
@@ -3286,69 +3286,6 @@ namespace GridTools
...
-   *
-   * This class facilitates the transfer by providing the save/load functions
-   * that are able to pack up the vector of CellId's and the associated
-   * data of type @p T into a stream.
-   *
-   * Type @p T is assumed to be serializable by <code>boost::serialization</code> (for
-   * example <code>unsigned int</code> or <code>std::vector@<double@></code>).
-   */
-  template <int dim, typename T>
-  struct CellDataTransferBuffer
```
Comment (peterrum) [inline include/deal.II/grid/grid_tools.h:3302]: Shouldn't we deprecate this class first?

Comment (kronbichler) [reply include/deal.II/grid/grid_tools.h:3302]: That's exactly what I wanted to ask here, I am unsure. But we never use it (now), we do not test it explicitly, and have no other reference to it.

**Comment by bangerth** [comment]
The function is in the public namespace. Let's deprecate the function first if you don't mind.

**Comment by bangerth** [comment]
The deprecation can probably still go into 9.4.

**Comment by kronbichler** [comment]
I see we have a clear opinion towards deprecation first.

**Pull request closed without merging**